src/users/impersonatelogin.php

```diff
@@ -0,0 +1,233 @@
+<?php
+/**
+ * @package     API
+ * @subpackage  plg_api_users
+ *
+ * @author      Techjoomla <extensions@techjoomla.com>
+ * @copyright   Copyright (C) 2009 - 2022 Techjoomla, Tekdi Technologies Pvt. Ltd. All rights reserved.
+ * @license     GNU GPLv2 <http://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
+ */
+
+// No direct access.
+defined('_JEXEC') or die('Restricted access');
+
+require_once JPATH_SITE . '/components/com_api/vendors/php-jwt/src/JWT.php';
+use Firebase\JWT\JWT;
+
+
+JModelLegacy::addIncludePath(JPATH_SITE . 'components/com_api/models');
+require_once JPATH_SITE . '/components/com_api/libraries/authentication/user.php';
+require_once JPATH_SITE . '/components/com_api/libraries/authentication/login.php';
+require_once JPATH_ADMINISTRATOR . '/components/com_api/models/key.php';
+require_once JPATH_ADMINISTRATOR . '/components/com_api/models/keys.php';
+
+/**
+ * Impersonate Login API resource class
+ *
+ * @package  API
+ * @since    1.6.0
+ */
+class UsersApiResourceImpersonateLogin extends ApiResource
+{
+	/**
+	 * Get method
+	 *
+	 * @return  object
+	 */
+	public function get()
+	{
+		$this->plugin->setResponse(JText::_('PLG_API_USERS_GET_METHOD_NOT_ALLOWED_MESSAGE'));
+	}
+
+	/**
+	 * Post method
+	 *
+	 * @return  object
+	 */
+	public function post()
+	{
+		$this->plugin->setResponse($this->keygen());
+	}
+
+	/**
+	 * Generate key method
+	 *
+	 * @return  object|boolean
+	 */
+	public function keygen()
+	{
+		// Init variables
+		$obj              = new stdclass;
+		$jinput           = JFactory::getApplication()->input;
+		$xImpersonate     = $jinput->server->get('X-Impersonate', '', 'STRING');
+		$httpXImpersonate = $jinput->server->get('HTTP_X_IMPERSONATE', '', 'STRING');
+
+		if (!empty($xImpersonate))
+		{
+			$userToImpersonate = $xImpersonate;
+		}
+		elseif (!empty($httpXImpersonate))
+		{
+			$userToImpersonate = $httpXImpersonate;
+		}
+
+		if (preg_match('/email:(\S+)/', $userToImpersonate, $matches))
+		{
+			$userId = $matches[1];
+		}
+		elseif (preg_match('/username:(\S+)/', $userToImpersonate, $matches))
+		{
+			$userId = JUserHelper::getUserId($matches[1]);
+		}
+		elseif (is_numeric($userToImpersonate))
+		{
+			$userId = $userToImpersonate;
+		}
+		else
+		{
+			ApiError::raiseError("403", JText::_('PLG_API_USERS_BAD_REQUEST_MESSAGE'));
+
+			return false;
+		}
+
+		if ($userId == null)
+		{
+			$model  = FD::model('Users');
```

Comment:
FD is older library of Easysocial.

WIll this work without ES when email is passed?

Reply:
No...I have added another function to fetch email